web/plugin-manager/src/components/pluginBlock.tsx

```diff
@@ -0,0 +1,183 @@
+        // Update data
+        setStatus(statusHelper(resultPluginInfo.data));
+
+        let result: AxiosResponse<number>;
+        if (!toggleStatus) {
+            // Launch plugin
+            try {
+                result = await axiosServices.manageLifePlugins(dataMemoized.id, "start");
+                // dataMemoized.status = PluginStatus.Up;
+                setStatus(!toggleStatus);
+                forcePlayStatus(true)
+                return result;
+            } catch (error) {
+                sendErrorData("error", "Start plugin failed");
+                // dataMemoized.status = PluginStatus.Error;
+                return;
+            }
+        } else {
+            // Stop plugin
+            try {
+                result = await axiosServices.manageLifePlugins(dataMemoized.id, "stop");
+                // dataMemoized.status = PluginStatus.Down;
+                forcePlayStatus(false)
+                setStatus(!toggleStatus);
+                return result;
+            } catch (error) {
+                sendErrorData("error", "Stop plugin failed");
+                // dataMemoized.status = PluginStatus.Error;
+                return;
+            }
+        }
+    }
+
+    // Update plugin
+    // Not implemented atm
+    function updatePlugins() {
+        //Front end update
+        // propsRef.current.updateAvailable = !propsRef.current.updateAvailable
+
+        //TODO : Uncoment this when we have a update process
+        //#####################################################
+        // const result = axiosServices.uploadPlugins("filename");
+        // if ( pluginsInfos.status && (pluginsInfos.status <= 200 || pluginsInfos.status >= 300)){
+        //     sendErrorData("error","Plugins infos failed to launch")
+        // }
+        // propsRef.current.online = true
+        // return result;
+        //#####################################################
+        console.log("Update is Not implemented ATM");
+    }
+    // Open plugin homepage
+    function openHomepagePlugins() {
+        if (statusHelper(dataMemoized.status)) window.open(dataMemoized.home);
+        else {
+            sendErrorData("error", "Plugin is not running can't be launched , launch it first");
+        }
+    }
+    // Uninstall plugin
+    function removePlugins() {
+        try {
+            axiosServices.deletePlugins(dataMemoized.id);
+            p.triggerRefreshPluginList();
+            sendErrorData("success", "Plugin removed");
+        } catch (error) {
+            sendErrorData("error", "Plugins failed to be removed");
+        }
+    }
+    // Minimize string to fit in the block
+    function minimizeString(str: string, length: number) {
+        if (str.length > length) {
+            return str.substring(0, length) + "...";
+        } else {
+            return str;
+        }
+    }
+
+    // Change the play status icon color and update the status if we want to force it
+    function playStatus() {
```

Comment:
This function name is misleading : if I read it without looking at the code I have the feeling that you will play a status.

Reply:
changed for : definePlayStatus